BeardedSpice/Tabs/VLCTabAdapter.m

```diff
@@ -0,0 +1,141 @@
+//
+//  VLCTabAdapter.m
+//  BeardedSpice
+//
+//  Created by Max Borghino on 2106-03-06
+//  Copyright (c) 2015 Tyler Rhodes / Jose Falcon. All rights reserved.
+//
+
+#import "VLCTabAdapter.h"
+#import "VLC.h"
+#import "runningSBApplication.h"
+#import "NSString+Utils.h"
+#import "MediaStrategy.h"
+
+#define APPNAME         @"VLC"
+#define APPID           @"org.videolan.vlc"
+
+@implementation VLCTabAdapter
+
++ (NSString *)displayName{
+
+    return APPNAME;
+}
+
++ (NSString *)bundleId{
+
+    return APPID;
+}
+
+- (NSString *)title {
+
+    @autoreleasepool {
+        VLCApplication *vlc = (VLCApplication *)[self.application sbApplication];
+        NSString *title;
+        if (![NSString isNullOrEmpty:vlc.nameOfCurrentItem])
+            title = vlc.nameOfCurrentItem;
+
+        if ([NSString isNullOrEmpty:title]) {
+            title = NSLocalizedString(@"No Track", @"SpotifyTabAdapter");
+        }
+
+        return [NSString stringWithFormat:@"%@ (%@)", title, APPNAME];
+    }
+}
+- (NSString *)URL{
+
+    return @"VLC";
+}
+
+// We have only one window.
+- (NSString *)key{
+
+    return @"A:VLC";
+}
+
+// We have only one window.
+-(BOOL) isEqual:(__autoreleasing id)otherTab{
+
+    if (otherTab == nil || ![otherTab isKindOfClass:[VLCTabAdapter class]]) return NO;
+
+    return YES;
+}
+
+// try to find the document which will respond to controls
+// NOTE: this feels brittle, it seems there should be a better way
+- (VLCDocument*) activeDocument {
+    VLCDocument *doc = NULL;
+    VLCApplication *vlc = (VLCApplication *)[self.application sbApplication];
+    if (vlc) {
+        NSEnumerator *e = [[vlc windows] objectEnumerator];
+        id o;
+        while ((o = [e nextObject]) && doc == NULL) {
+            NSLog(@"VLC window %@: %lu: %@", o, [o index], [o name]);
+            if ([[o name] isEqualToString: [vlc nameOfCurrentItem]]) {
+                doc = [[[vlc windows] objectAtIndex: [o index]] document];
+            }
+        }
+    }
+    return doc;
+}
+
+//////////////////////////////////////////////////////////////
+#pragma mark Player control methods
+//////////////////////////////////////////////////////////////
+
+- (void)toggle{
+    VLCDocument *doc = [self activeDocument];
+    if (doc) {
+        [doc play];
+    }
+}
+
+- (void)pause{
+    VLCApplication *vlc = (VLCApplication *)[self.application sbApplication];
+    if (vlc && [vlc playing]) {
+        VLCDocument *doc = [self activeDocument];
+        if (doc) {
+            [doc play];
```

Comment:
@fmborghino You guessed right. [In my question](http://stackoverflow.com/q/35670703), I was trying to use `[vlc stop]` to pause, which was bad for a number of reasons:

1. It actually closed the file.
2. This caused it to hang when resumed.
3. For some reason (bug?), `[vlc stop]` is completely unavailable, along with some other methods. However, it is available [by using plain Applescript](http://stackoverflow.com/a/35774824/3398839), which I was [trying to avoid](http://stackoverflow.com/questions/35670703/applescript-vlc-scripting-bridge#comment59241547_35774824).

It never occurred to toggle it with `play` (most applications like VLC have separate play and pause methods). Also, my implementation only worked if there was one window. I admit, you beat me to it with a much better implementation. :disappointed: